tests/TestClassRedirectIfOldUrl.php

```diff
@@ -97,39 +101,27 @@ public function testRedirectDetailViewSameUrl()
 	}
 
 	/**
-	 * @covers \onOffice\WPlugin\Utility\RedirectIfOldUrl::redirectDetailView
+	 * @covers \onOffice\WPlugin\Utility\Redirector::redirectDetailView
 	 */
 	public function testRedirectDetailViewSameUrlWithoutOption()
 	{
 		global $wp;
 		$wp->request = 'detail-view/3333';
-		$wpPageWrapper = $this->getMockBuilder(WPPageWrapper::class)
-			->setMethods(['getPageLinkByPageId', 'getPageUriByPageId'])
-			->getMock();
-
-		$wpPageWrapper->method('getPageLinkByPageId')->willReturn('https://abc.com');
-		$wpPageWrapper->method('getPageUriByPageId')->willReturn('test-net');
-
-		$pLanguageSwitcher = $this->getMockBuilder(EstateDetailUrl::class)
-			->setMethods(['createEstateDetailLink'])
-			->getMock();
-		$pLanguageSwitcher->method('createEstateDetailLink')->willReturn('https://abc.com/detail/33-test-post');
-
-		$wpRedirectWrapper = $this->getMockBuilder(WPRedirectWrapper::class)
-			->setMethods(['redirect'])
-			->getMock();
-		$wpRedirectWrapper->method('redirect')->willReturn(true);
-
-		 $redirectIfOldUrl = new RedirectIfOldUrl();
-		 $redirectIfOldUrl->setPLanguageSwitcher($pLanguageSwitcher);
-		 $redirectIfOldUrl->setWpRedirectWrapper($wpRedirectWrapper);
-		 $redirectIfOldUrl->setWpPageWrapper($wpPageWrapper);
-
-		 $this->assertTrue($redirectIfOldUrl->redirectDetailView(1, 1, 'tes post'));
+		$pWPPost = self::factory()->post->create_and_get([
+			'post_author' => 1,
+			'post_content' => '[oo_estate view="detail"]',
+			'post_title' => 'Detail View',
+			'post_type' => 'page',
+		]);
+		$pLanguageSwitcher = new EstateDetailUrl();
+		$wpRedirectWrapperMocker = new RedirectWrapperMocker();
```

Comment:
Mock could be built using PHPUnit, instead of making a new class (which should also be tested)

Reply:
@jayay Please review the [above](https://github.com/onOfficeGmbH/oo-wp-plugin/pull/132#discussion_r776669095). Because you asked me to write Mocker Class

Reply:
Either way works